Integrations/EDL/EDL.py

```diff
@@ -0,0 +1,371 @@
+#     while len(iocs) != pre_consolidate_iocs_len:
+#         pre_consolidate_iocs_len = len(iocs)
+#         iocs, next_page = find_new_ips_and_consolidate(indicator_query, iocs, limit, next_page)
+#     return iocs
+#
+#
+# def find_new_ips_and_consolidate(indicator_query, iocs, limit, next_page=0):
+#     """
+#     Finds new IPs and consolidates them
+#     """
+#     last_iocs_found, next_page = find_indicators_to_limit_loop(indicator_query, limit, total_fetched=len(iocs),
+#                                                                next_page=next_page)
+#     iocs = consolidate_ips(iocs)
+#     return iocs, next_page
+#
+#
+# def consolidate_ips(iocs):
+#     """
+#     Groups together ips in a list of strings
+#     """
+#     try:
+#         iocs = sorted(iocs, key=lambda ip: struct.unpack('!L', inet_aton(ip))[0])
+#     except OSError:
+#         demisto.debug('Failed to consolidate IPs')
+#         return iocs
+
+
+def create_csv_out_list(cache_dict: dict) -> list:
+    """
+    Creates a csv output result
+    """
+    csv_headers = cache_dict.pop(CSV_FIRST_LINE_KEY, '')
+    values_list = list(cache_dict.values())
+    if csv_headers:
+        values_list.insert(0, csv_headers)
+
+    return values_list
+
+
+def create_values_out_dict(iocs: list, out_format: str) -> dict:
+    """
+    Create a dictionary for output values using the selected format
+    """
+    out_format_func = {
+        FORMAT_TEXT: format_ioc_text_out,
+        FORMAT_JSON_SEQ: format_ioc_json_seq_out,
+        FORMAT_CSV: format_ioc_csv_out
+    }
+    return create_formatted_values_out_dict(iocs, out_format, out_format_func.get(out_format, str))
+
+
+def create_formatted_values_out_dict(iocs: list, out_format: str, out_format_func: Callable) -> dict:
+    """
+    Create a dictionary for output values formatted in the selected out_format
+    """
+    out_dict = {}
+    if out_format == FORMAT_JSON:
+        iocs_list = [ioc for ioc in iocs]
+        return {'iocs_list': json.dumps(iocs_list, indent=4)}
+    else:
+        for ioc in iocs:
+            value = ioc.get('value')
+            if value:
+                out_dict[value] = out_format_func(ioc)
+        if out_format == 'csv' and len(iocs) > 0:
+            # add csv headers to out_dict under saved key
+            headers = list(iocs[0].keys())
+            out_dict[CSV_FIRST_LINE_KEY] = list_to_str(headers)
+        return out_dict
+
+
+def format_ioc_text_out(ioc: dict) -> str:
+    """
+    Return output in text format
+    """
+    return ioc.get('value', '')
+
+
+def format_ioc_json_seq_out(ioc: dict) -> str:
+    """
+    Return output in json seq format
+    """
+    return json.dumps(ioc)
+
+
+def format_ioc_csv_out(ioc: dict) -> str:
+    """
+    Return output in csv format
+    """
+    values = list(ioc.values())
+    # wrap csv values with " to separate them
+    return list_to_str(values, map_func=lambda val: f'"{val}"')
```

Comment:
we can remove all this and replace it with 7 lines switch. see my comment at line 166